lua/octo/model/pull-request.lua

```diff
@@ -172,4 +172,31 @@ function PullRequest:get_commit_changed_files(rev, callback)
   }
 end
 
+---Helper function to create a PullRequest with merge base
+---@param opts table Options for PullRequest creation
+---@param obj table PR object from GraphQL response
+---@param cb function Callback function(pull_request)
+function M.create_with_merge_base(opts, obj, cb)
+  local Rev = require("octo.reviews.rev").Rev
+  local owner, name = utils.split_repo(opts.repo)
+
+  -- Fetch merge base for accurate diff
+  gh.fetch_merge_base(owner, name, obj.baseRefName, obj.headRefName, function(merge_base_sha, stderr)
+    if stderr and not utils.is_blank(stderr) then
```

Comment:
The `gh.create_callback { success = ..., failure = ... }` could be useful here

Reference: https://github.com/pwntester/octo.nvim/discussions/876